### DISCUSSION TITLE
1.0.0

### DISCUSSION THREAD

**Discussion by kairoh** [kairoh/studio#29] [Announcements]
**Major changes**

* Full rewrite based on Quarkus Framework
* Add native edition (=native executables for 64bits Linux, Windows and MacOS)
* Use current directory by default (=portable), thanks to @Steph29
* Add USB support for MacOS M1 (usb4java for aarch64) : #11 
* Fix MacOS Ventura FS support (fix #12 )
* Optimize multithreading support
* Greatly improve code coverage (from 10% to 52%)

**Minor changes**

* Add devops with Github actions
* Use Lombok beans
* Replace Vorbis-java.jar by JVorbis.jar (pure java lib)
* Replace Gson by Jackson
* Merge project *metadata* into *web-ui*
* Each story have an UUID (use 1st node UUID if missing)

<hr /><em>This discussion was created from the release <a href='https://github.com/kairoh/studio/releases/tag/1.0.0'>1.0.0</a>.</em>

**Comment by JulienMaille** [comment]
Thanks for the release, one remark though, in the windows archive, the .bat file contains an invalid filename:
```
:: batch args as command args
%CWD%\studio*runner.exe %*
```
instead of
>studio-web-ui-1.0.0-runner

**Reply by NicoLeOca** [reply]
isn't this designed to include all files starting by "studio" and ending with "runner", by using a star "*"?

**Reply by JulienMaille** [reply]
Could be, but does that work?

**Reply by kairoh** [reply]
Nope it doesn't...
But it's fixed in  <a href='https://github.com/kairoh/studio/releases/tag/1.0.1'>1.0.1</a> !

**Comment by NicoLeOca** [comment]
Salut @kairoh 
j'ai un soucis de droits sur MacOS, que ce soit en SU, ou avec mon user.
Saurais-tu de quoi ça vient?
Je n'avais pas ces soucis sur la release précédente

```
studio.sh: line 9: /Users/moi/Downloads/studio-1.0.0-macOS/studio-web-ui-1.0.0-runner: Permission denied
studio.sh: line 9: exec: /Users/moi/Downloads/studio-1.0.0-macOS/studio-web-ui-1.0.0-runner: cannot execute: Undefined error: 0
```

La version Java se lance quant à elle. Et je confirme que ce fix fonctionne https://github.com/kairoh/studio/issues/12

**Reply by Seph29** [reply]
> Salut @kairoh j'ai un soucis de droits sur MacOS, que ce soit en SU, ou avec mon user. Saurais-tu de quoi ça vient? Je n'avais pas ces soucis sur la release précédente
> 
> ```
> studio.sh: line 9: /Users/moi/Downloads/studio-1.0.0-macOS/studio-web-ui-1.0.0-runner: Permission denied
> studio.sh: line 9: exec: /Users/moi/Downloads/studio-1.0.0-macOS/studio-web-ui-1.0.0-runner: cannot execute: Undefined error: 0
> ```
> 
> La version Java se lance quant à elle. Et je confirme que ce fix fonctionne #12

Tu peux tester :
chmod +x studio-web-ui-1.0.0-runner
chmod +x studio.sh

**Reply by NicoLeOca** [reply]
Merci pour le retour @Seph29 
Je l'avais fait sur le runner mais pas sur le shell

j'ai cette erreur maintenant: 
```
zsh: segmentation fault  ./studio.sh
```

saurais-tu d'où ça vient?
Quelques recherches me disent que ça vient du code.
as-tu réussi à lancer cette release de ton coté?

**Reply by Seph29** [reply]
Je viens de faire le test sur ma VM sous MacOs 13 même erreur
par contre c'est Ok sur ma VM sous MacOs 12

**Reply by kairoh** [reply]
Je buildais sur MacOS 11, a priori compatible 11 et 12.
Je suis passé à runner Gitlab en MacOS 12, a priori compatible MacOs 12 et 13.
A tester avec la  <a href='https://github.com/kairoh/studio/releases/tag/1.0.1'>1.0.1</a>

**Reply by Seph29** [reply]
@kairoh je te refais une fournée de test ce soir 😄

**Reply by NicoLeOca** [reply]
Bonjour @kairoh et merci pour les fix
Sur MacOS 13, je n'ai plus les soucis de droits.
Par contre, j'ai toujours ceci
```
zsh: segmentation fault  ./studio.sh
```

**Reply by Seph29** [reply]
Pareil pour moi

**Reply by kairoh** [reply]
Votre MacOS 13, il a un CPU x86_64 ou aarch64?
Vous n'avez plus de trace d'erreur? (segfault c'est un peu un fourre-tout)

**Reply by Seph29** [reply]
```
seph29@Seph29s-iMac-Pro studio-1.0.1-MacOS % lldb --file studio-web-ui-1.0.1-runner
(lldb) target create "studio-web-ui-1.0.1-runner"
Current executable set to '/Users/seph29/Downloads/studio-1.0.1-MacOS/studio-web-ui-1.0.1-runner' (x86_64).
(lldb) r
Process 911 launched: '/Users/seph29/Downloads/studio-1.0.1-MacOS/studio-web-ui-1.0.1-runner' (x86_64)
Process 911 stopped
* thread #1, stop reason = EXC_BAD_ACCESS (code=1, address=0xc8)
    frame #0: 0x00007ff81b2a3ba5
->  0x7ff81b2a3ba5: movl   0xc8(%rbx), %eax
    0x7ff81b2a3bab: testl  %eax, %eax
    0x7ff81b2a3bad: je     0x7ff81b2a3bb8
    0x7ff81b2a3baf: movq   (%r15), %rcx
Target 0: (studio-web-ui-1.0.1-runner) stopped.```

**Comment by Makinitz** [comment]
Hello
Un grand merci.
Je n'arrive pas toutefois à faire détecter ma Lunii par Studio. 
Tout se lance bien, mes packs d'histoires sont bien chargés. La Lunii est bien vue par l'application officielle LuniiStore. Mais impossible de la visualiser dans Studio pour transférer des histoires.

2023-01-04 15:29:21,576 INFO  [stu.web.Studio] (main) Running Studio
2023-01-04 15:29:21,982 INFO  [stu.web.api.SockJsEventBusBridge] (main) Start Sockjs EventBus
2023-01-04 15:29:21,985 INFO  [stu.web.ser.LibraryService] (main) library path : .\library (tmpdir path : .\tmp)
2023-01-04 15:29:22,005 INFO  [stu.met.DatabaseMetadataService] (main) Read database .\db\unofficial.json
2023-01-04 15:29:22,026 INFO  [stu.met.DatabaseMetadataService] (main) Read database .\db\official.json
2023-01-04 15:29:22,042 INFO  [stu.met.DatabaseMetadataService] (main) Clean official packs in library database
2023-01-04 15:29:22,043 INFO  [stu.web.Studio] (main) Opening URL http://localhost:8080/ in default browser...
2023-01-04 15:29:22,086 INFO  [io.quarkus] (main) studio-web-ui 1.0.0 native (powered by Quarkus 2.14.2.Final) started in 0.507s. Listening on: http://localhost:8080
2023-01-04 15:29:22,086 INFO  [io.quarkus] (main) Profile prod activated.
2023-01-04 15:29:22,088 INFO  [io.quarkus] (main) Installed features: [awt, cdi, rest-client-reactive, rest-client-reactive-jackson, resteasy-reactive, resteasy-reactive-jackson, smallrye-context-propagation, usb4java, vertx]
2023-01-04 15:29:26,114 INFO  [stu.web.ser.StoryTellerService] (vert.x-eventloop-thread-0) Setting up story teller driver
2023-01-04 15:29:26,116 INFO  [stu.dri.usb.LibUsbDetectionHelper] (vert.x-eventloop-thread-0) Initializing libusb...
2023-01-04 15:29:26,134 INFO  [stu.dri.usb.LibUsbDetectionHelper] (vert.x-eventloop-thread-0) Hotplug is NOT supported. Scheduling task to actively poll USB device...
2023-01-04 15:29:26,135 INFO  [stu.dri.usb.LibUsbDetectionHelper] (vert.x-eventloop-thread-0) Initializing libusb...
2023-01-04 15:29:26,135 INFO  [stu.dri.usb.LibUsbDetectionHelper] (vert.x-eventloop-thread-0) Hotplug is NOT supported. Scheduling task to actively poll USB device...
2023-01-04 15:29:26,178 INFO  [stu.web.ser.LibraryService] (executor-thread-0) Read metadata ARCHIVE from pack: 3 ]Didier_Jeunesse_-_Casse-Noisette_-_Conte musical_(22 pistes_48mn)[by_Gayel-1670052283958.zip

**Reply by kairoh** [reply]
Il y avait une régression dans la détection USB sur Windows (polling non lancé).
C'est corrigé dans la <a href='https://github.com/kairoh/studio/releases/tag/1.0.1'>1.0.1</a>

**Comment by taker24** [comment]
Bonjour, j'ai le meme soucis que @Makinitz   ma lunii (version 2 maj 2.22) n'est pas detectee par studio sous windows 11 et quand je lance sutdio j'ai un message d'erreur m'ndiquent "echec de recupération des packs" et quand j'essayer dans charqer un j'ai un petit encart blanc qui indique " televersement du pack" (ou quelque chose dans le genre) mais les telechargement ne progresse pas. 
Avez-vous une solution ? 
Merci d'avance

**Reply by kairoh** [reply]
Réessaie avec la  <a href='https://github.com/kairoh/studio/releases/tag/1.0.1'>1.0.1</a>
Pour les autres erreurs, il y a surement d'autres soucis (mauvais chemin de librairie ou pack invalide).

**Reply by taker24** [reply]
bonjour kairoh et merci pour ta réponse rapide,
la nouvelle maj a corriger le problème de détection de la lunii maintenant studio la voit et voit les histoire dedans.
j'ai toujours le problème de transfert de l'histoire qui ce lance et ne progresse jamais. studio m'indique Bibliothèque locale 
Chemin : .\library peut-être que ce viens de là mais je ne sais pas comment le modifier. j'ai juste dézipper ta release dans un dossier sur le bureau
